src/RealFourierArray.jl

```diff
@@ -0,0 +1,91 @@
+# Lazy structure for an array whose real and fourier part can be accessed by A.real and A.fourier
+# For normalization conventions, see PlaneWaveBasis.jl
+
+import Base.getproperty, Base.setproperty!, Base.fieldnames
+import Base.*, Base.+, Base./
+import Base.eltype
+
+## these functions are minimally optimized for the moment
+
+## TODO this always allocates both real and fourier parts, which might be a bit wasteful
+## TODO implement broadcasting. This seems to be non-trivial, so I just implement a few simple methods
+## TODO implement compressed representation in Fourier space of real arrays
```

Comment:
That's for in-place, we can just use out-of-place. The thing that makes me hesitate is that the compressed representation isn't unitary (eg `norm(f_fourier)` is not the same in both representations), and it's not even a constant factor, so that could lead to subtle bugs. Anyway, that's a factor-of-two optimization, so low-priority.

Reply:
Really? I mean you can always overload functions appropriately, right? But I agree with low priority here.